### ISSUE TITLE
[Data Views][Accessibility] Improvements to grid layout semantics/navigation

### ISSUE THREAD

**Issue by andrewhayward** [WordPress/gutenberg#59188]
## What problem does this address?

- Grid layouts are currently semantically ambiguous, with no structural scaffolding to denote which fields relate to each other.
- As such, they are also difficult to navigate with a keyboard, as every tab stop/element is presented to the user with no way to skip content.
- Media items are also rendered as buttons, but with no clear semantic purpose as to what they do.
- Grid items can be selected/deselected with a <kbd>Cmd</kbd>/<kbd>Ctrl</kbd> click, but this is not apparent until the click happens.

## What is your proposed solution?

- [x] Add structural semantics to grid items
  - At a minimum, each item should be marked as a [`group`](https://w3c.github.io/aria/#group). However, given the 'grid' aesthetic of the layout, a [`grid`](https://w3c.github.io/aria/#grid)/[`gridcell`](https://w3c.github.io/aria/#gridcell) relationship would be more appropriate.
  - Note that grid cells _must_ be appropriately contained by [rows](https://w3c.github.io/aria/#row).
  - **https://github.com/WordPress/gutenberg/pull/59279**
- [x] Improve keyboard navigation
  - Allow users to navigate through grid items without having to hit every tab stop (for example, by using [a grid pattern](https://www.w3.org/WAI/ARIA/apg/patterns/grid/)).
  - **https://github.com/WordPress/gutenberg/pull/59279**
- [ ] Remove media item buttons, and instead allow primary link actions to consume their footprint.
  - This might be by including the media item in the link, wrapping the media item in an equivalent link, or by covering the footprint with the primary link.
- [x] Make it clearer that <kbd>Cmd</kbd>/<kbd>Ctrl</kbd> clicking a grid item will select/deselect it
  - This could be by changing the cursor during hover, or changing the state of the grid item
  - **https://github.com/WordPress/gutenberg/pull/59292**

**Comment by youknowriad** [comment]
I think this is mostly solved, there's a dedicated issue for the keyboard navigation #59188
